doc/source/development/rfc/rfc94_field_precision_width_metadata.rst

```diff
@@ -0,0 +1,173 @@
+.. _rfc-94:
+
+=============================================================
+RFC 94: Numeric fields width/precision metadata
+=============================================================
+
+============== =============================================
+Author:        Alessandro Pasotti
+Contact:       elpaso @ itopen.it
+Started:       2023-Feb-17
+Status:        Development
+Target:        GDAL 3.7
+============== =============================================
+
+Summary
+-------
+
+The document proposes and describes the introduction of a new method
+to the `OGRFieldDefn` class that returns information for numeric real
+fields about how the precision and width have been calculated by the
+driver and have to be interpreted.
+
+Motivation
+----------
+
+Applications (for example OGR/GDAL utils and QGIS) may require information
+about the field width and precision in order to convert between different
+formats or to validate user input on editing.
+
+Different drivers may calculate the width and precision differently by including
+or not the decimal separator and/or the minus sign into the calculated length.
+
+There is currently no way for an application to access this information and this
+can lead to loss of width or precision while converting between formats or when the
+application decides to stay on the safe side and reduce the width reported by GDAL.
+
+Additionally, attention must be paid to the meaning of "width" and "precision":
+OGR "width" corresponds to SQL "precision" and OGR "precision" corresponds to SQL "scale".
+
+
+For reference:
+
+- https://trac.osgeo.org/gdal/ticket/6960
+- https://issues.qgis.org/issues/11755
+- https://issues.qgis.org/issues/15188#note-8
+- https://github.com/qgis/QGIS/issues/51849
+
+
+Current drivers behavior
+------------------------
+
+Here is a list of the drivers and how they behave with respect to the width and precision,
+(for databases we refer to the `NUMERIC` data type).
+
+================== ================================== =====================
+ Driver             Width Includes Decimal Separator   Width Includes Sign
+================== ================================== =====================
+ Shapefile          YES                                YES
+ MapInfo            ?                                  ?
```

Comment:
TABDATFile::WriteDecimalField()
```
    snprintf(szFormat, sizeof(szFormat), "%%%d.%df", nWidth, nPrec);
    const char *pszVal = CPLSPrintf(szFormat, dValue);
    if (static_cast<int>(strlen(pszVal)) > nWidth)
    {
        CPLError(CE_Failure, CPLE_AppDefined,
                 "Cannot format %g as a %d.%d field", dValue, nWidth, nPrec);
        return -1;
    }
```
tends to show it is the same as shapefile. The .dat file format is very similar to .dbf
Also see https://trac.osgeo.org/gdal/ticket/6392 and https://docs.safe.com/fme/html/FME_Desktop_Documentation/FME_ReadersWriters/mif/user-attributes.htm